### PR TITLE
[YUNIKORN-2658]add nolint:funlen to long functions to supress the lint warnings

### PR DESCRIPTION
**Pull request by rich7420**
### What is this PR for?

To add nolint:funlen to long functions to supress the lint warnings, when we use `make lint` 
like this 
image

### What type of PR is it?

- [x] - Improvement

### Todos
-  Improve other warn message

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-2658

### How should this be tested?
in Makefile take off `--new-from-rev=$${headSHA}`
then run `make lint
`
### Screenshots (if appropriate)
NA
### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/scheduler/objects/application_state.go [hunk 1]
```diff
@@ -77,6 +77,7 @@ func (as applicationState) String() string {
 	return [...]string{"New", "Accepted", "Starting", "Running", "Rejected", "Completing", "Completed", "Failing", "Failed", "Expired", "Resuming"}[as]
 }
 
+//nolint:funlen
 func NewAppState() *fsm.FSM {
```
Comment (chia7712) [inline pkg/scheduler/objects/application_state.go:81]: could you please do the similar refactor (https://github.com/apache/yunikorn-k8shim/pull/861)?

File: pkg/scheduler/partition.go [hunk 2]
```diff
@@ -1207,6 +1207,8 @@ func (pc *PartitionContext) calculateNodesResourceUsage() map[string][]int {
 
 // removeAllocation removes the referenced allocation(s) from the applications and nodes
 // NOTE: this is a lock free call. It must NOT be called holding the PartitionContext lock.
+//
+//nolint:funlen
 func (pc *PartitionContext) removeAllocation(release *si.AllocationRelease) ([]*objects.Allocation, *objects.Allocation) {
```
Comment (chia7712) [inline pkg/scheduler/partition.go:1212]: It seems we can extract the code of generating `released` into a separate method. WDYT?

**Review by chia7712 — COMMENTED** [review]
@holdenk thanks for this patch

File: pkg/scheduler/objects/application_state.go (hunk 1 above)
Comment (rich7420) [reply pkg/scheduler/objects/application_state.go:81]: Yes,no problem!

**Comment by chia7712** [comment]
@rich7420 could you please fix conflicts?

**Comment by rich7420** [comment]
@chia7712 Thx, I finished the conflict.

File: pkg/scheduler/partition.go [hunk 3]
```diff
@@ -1241,54 +1241,55 @@ func (pc *PartitionContext) calculateNodesResourceUsage() map[string][]int {
...
-	}
-	// Processing a removal while in the Completing state could race with the state change.
-	// The race occurs between removing the allocation and updating the queue after node processing.
-	// If the state change removes the queue link before we get to updating the queue after the node we
-	// leave the resources as allocated on the queue. The queue cannot be removed yet at this point as
-	// there are still allocations left. So retrieve the queue early to sidestep the race.
-	queue := app.GetQueue()
-	// temp store for allocations manipulated
+//
+//nolint:funlen
```
Comment (chia7712) [inline pkg/scheduler/partition.go:1245]: this can be removed now, right?

File: pkg/scheduler/partition.go [hunk 4]
```diff
@@ -1241,54 +1241,55 @@ func (pc *PartitionContext) calculateNodesResourceUsage() map[string][]int {
 
 // removeAllocation removes the referenced allocation(s) from the applications and nodes
```
Comment (chia7712) [inline pkg/scheduler/partition.go:1242]: please move this comment to `removeAllocation`

**Comment by chia7712** [comment]
@rich7420 Could you please fix the error? You can run `make test` to make sure everything is alright

**Comment by rich7420** [comment]
Ok! I will finish it today.

File: pkg/scheduler/partition.go [hunk 5]
```diff
@@ -1245,23 +1245,23 @@ func (pc *PartitionContext) calculateNodesResourceUsage() map[string][]int {
 //nolint:funlen
 func (pc *PartitionContext) generateReleased(release *si.AllocationRelease, app *objects.Application) []*objects.Allocation {
 	released := make([]*objects.Allocation, 0)
-	uuid := release.GetUUID()
-	if uuid == "" {
-		log.Logger().Info("remove all allocations",
+	allocationKey := release.GetAllocationKey() // 获取 allocationKey，而不是 UUID
```
Comment (pbacsko) [inline pkg/scheduler/partition.go:1248]: Nit: please use english comments if necessary

**Review by pbacsko — CHANGES_REQUESTED** [review]
One final comment.

**Review by pbacsko — COMMENTED** [review]
Please address the comments from @chia7712 otherwise it looks good.

File: pkg/scheduler/objects/application_state.go [hunk 6]
```diff
@@ -90,141 +90,150 @@ func (as applicationState) String() string {
...
-			fmt.Sprintf("leave_%s", Running.String()): func(_ context.Context, event *fsm.Event) {
-				if event.Dst != Running.String() {
-					app := event.Args[0].(*Application) //nolint:errcheck
-					app.queue.decRunningApps()
-					metrics.GetQueueMetrics(app.queuePath).DecQueueApplicationsRunning()
-					metrics.GetSchedulerMetrics().DecTotalApplicationsRunning()
-				}
-			},
-			fmt.Sprintf("enter_%s", Completed.String()): func(_ context.Context, event *fsm.Event) {
+//nolint:funlen
```
Comment (chia7712) [inline pkg/scheduler/objects/application_state.go:93]: this is unnecessary now, right?

**Review by chia7712 — COMMENTED** [review]
@rich7420 Could you please fix following cases?
```
pkg/scheduler/placement/fixed_rule_test.go:127: Function 'TestFixedRuleParent' is too long (146 > 120) (funlen)
func TestFixedRuleParent(t *testing.T) {
pkg/scheduler/placement/provided_rule_test.go:31: Function 'TestProvidedRulePlace' is too long (124 > 120) (funlen)
func TestProvidedRulePlace(t *testing.T) {
pkg/scheduler/placement/provided_rule_test.go:158: Function 'TestProvidedRuleParent' is too long (158 > 120) (funlen)
func TestProvidedRuleParent(t *testing.T) {
pkg/scheduler/placement/tag_rule_test.go:63: Function 'TestTagRulePlace' is too long (133 > 120) (funlen)
func TestTagRulePlace(t *testing.T) {
pkg/scheduler/placement/tag_rule_test.go:199: Function 'TestTagRuleParent' is too long (144 > 120) (funlen)
func TestTagRuleParent(t *testing.T) {
pkg/scheduler/placement/user_rule_test.go:90: Function 'TestUserRuleParent' is too long (142 > 120) (funlen)
func TestUserRuleParent(t *testing.T) {
pkg/scheduler/objects/preemption.go:211: Function 'calculateVictimsByNode' is too long (133 > 120) (funlen)
func (p *Preemptor) calculateVictimsByNode(nodeAvailable *resources.Resource, potentialVictims []*Allocation) (int, []*Allocation) {
pkg/scheduler/partition_test.go:3850: Function 'TestUserHeadroom' has too many statements (89 > 80) (funlen)
func TestUserHeadroom(t *testing.T) {
pkg/scheduler/tests/application_tracking_test.go:48: Function 'TestApplicationHistoryTracking' is too long (124 > 120) (funlen)
```

File: pkg/scheduler/partition.go [hunk 7]
```diff
@@ -1239,56 +1239,55 @@ func (pc *PartitionContext) calculateNodesResourceUsage() map[string][]int {
...
-	var confirmed *objects.Allocation
-	// when allocationKey is not specified, remove all allocations from the app
+	allocationKey := release.GetAllocationKey()
 	if allocationKey == "" {
 		log.Log(log.SchedPartition).Info("remove all allocations",
-			zap.String("appID", appID))
+			zap.String("appID", app.ApplicationID))
 		released = append(released, app.RemoveAllAllocations()...)
 	} else {
-		// if we have an allocationKey the termination type is important
```
Comment (chia7712) [inline pkg/scheduler/partition.go:1274]: please keep this comment

File: pkg/scheduler/objects/application_state.go [hunk 8]
```diff
@@ -90,141 +90,149 @@ func (as applicationState) String() string {
...
-					app := event.Args[0].(*Application) //nolint:errcheck
-					app.queue.decRunningApps()
-					metrics.GetQueueMetrics(app.queuePath).DecQueueApplicationsRunning()
-					metrics.GetSchedulerMetrics().DecTotalApplicationsRunning()
-				}
-			},
-			fmt.Sprintf("enter_%s", Completed.String()): func(_ context.Context, event *fsm.Event) {
+func eventDesc() fsm.Events {
+	return fsm.Events{
+		{
```
Comment (chia7712) [inline pkg/scheduler/objects/application_state.go:95]: please keep the format to reduce the changes.
```go
			{
				Name: RejectApplication.String(),
				Src:  []string{New.String()},
				Dst:  Rejected.String(),
			}, {
				Name: RunApplication.String(),
				Src:  []string{New.String(), Resuming.String()},
				Dst:  Accepted.String(),
			}, {
				Name: RunApplication.String(),
				Src:  []string{Accepted.String(), Running.String(), Completing.String()},
				Dst:  Running.String(),
			}, {
				Name: CompleteApplication.String(),
				Src:  []string{Accepted.String(), Running.String()},
				Dst:  Completing.String(),
			}, {
				Name: CompleteApplication.String(),
				Src:  []string{Completing.String()},
				Dst:  Completed.String(),
			}, {
				Name: FailApplication.String(),
				Src:  []string{New.String(), Accepted.String(), Running.String()},
				Dst:  Failing.String(),
			}, {
				Name: FailApplication.String(),
				Src:  []string{Failing.String()},
				Dst:  Failed.String(),
			}, {
				Name: ResumeApplication.String(),
				Src:  []string{New.String(), Accepted.String()},
				Dst:  Resuming.String(),
			}, {
				Name: ExpireApplication.String(),
				Src:  []string{Completed.String(), Failed.String(), Rejected.String()},
				Dst:  Expired.String(),
			},
```

File: pkg/scheduler/partition.go [hunk 9]
```diff
@@ -1239,56 +1239,55 @@ func (pc *PartitionContext) calculateNodesResourceUsage() map[string][]int {
...
-	app := pc.getApplication(appID)
-	// no app nothing to do everything should already be clean
-	if app == nil {
-		log.Log(log.SchedPartition).Info("Application not found while releasing allocation",
-			zap.String("appID", appID),
-			zap.String("allocationKey", allocationKey),
-			zap.Stringer("terminationType", release.TerminationType))
-		return nil, nil
-	}
-	// Processing a removal while in the Completing state could race with the state change.
```
Comment (chia7712) [inline pkg/scheduler/partition.go:1259]: please keep it

File: pkg/scheduler/objects/application_state.go [hunk 10]
```diff
@@ -90,141 +90,149 @@ func (as applicationState) String() string {
...
-				Src:  []string{New.String(), Accepted.String()},
-				Dst:  Resuming.String(),
-			}, {
-				Name: ExpireApplication.String(),
-				Src:  []string{Completed.String(), Failed.String(), Rejected.String()},
-				Dst:  Expired.String(),
-			},
-		},
-		fsm.Callbacks{
-			// The state machine is tightly tied to the Application object.
```
Comment (chia7712) [inline pkg/scheduler/objects/application_state.go:135]: please keep it

**Review by chia7712 — COMMENTED** [review]
@rich7420 thanks for patch. Could you please DON"T remove comments? that could be useful in tracing code or a hint for debugging.

File: pkg/scheduler/objects/application_state.go [hunk 11]
```diff
@@ -90,141 +90,146 @@ func (as applicationState) String() string {
...
-					metrics.GetQueueMetrics(app.queuePath).DecQueueApplicationsRunning()
-					metrics.GetSchedulerMetrics().DecTotalApplicationsRunning()
-				}
-			},
-			fmt.Sprintf("enter_%s", Completed.String()): func(_ context.Context, event *fsm.Event) {
+	}
+}
+
+func callbacks() fsm.Callbacks {
+	// The state machine is tightly tied to the Application object.
```
Comment (chia7712) [inline pkg/scheduler/objects/application_state.go:136]: Could you please move the comments to line#134?

File: pkg/scheduler/partition.go [hunk 12]
```diff
@@ -1239,56 +1239,58 @@ func (pc *PartitionContext) calculateNodesResourceUsage() map[string][]int {
...
-	// The race occurs between removing the allocation and updating the queue after node processing.
-	// If the state change removes the queue link before we get to updating the queue after the node we
-	// leave the resources as allocated on the queue. The queue cannot be removed yet at this point as
-	// there are still allocations left. So retrieve the queue early to sidestep the race.
-	queue := app.GetQueue()
-	// temp store for allocations manipulated
+// temp store for allocations manipulated
```
Comment (chia7712) [inline pkg/scheduler/partition.go:1242]: this is duplicate as it is existent in line#1291

**Review by chia7712 — COMMENTED** [review]
@rich7420 thanks for updated patch

**Comment by rich7420** [comment]
@chia7712 I completed modifying the comments. Thank you for your feedback. Thank you very much!

**Pull request closed without merging**